server/src/test/java/org/elasticsearch/common/settings/AddStringKeyStoreCommandTests.java

```diff
@@ -133,6 +134,16 @@ public void testStdinLong() throws Exception {
         assertSecureString("foo", "secret value 2");
     }
 
+    public void testAddUtf8String() throws Exception {
+        KeyStoreWrapper.create().save(env.configFile(), new char[0]);
+        byte[] bytes = new byte[randomIntBetween(8, 16)];
+        new Random().nextBytes(bytes);
```

Comment:
This won't guarantee any codepoints above 127. I think we should always ensure that, so that this test cannot randomly pass if the copying code you fixed is broken?